praw/__init__.py

```diff
@@ -2569,28 +2583,36 @@ class ReportMixin(AuthenticatedReddit):
 
     @decorators.restrict_access(scope='report')
     def hide(self, thing_id, _unhide=False):
-        """Hide up to 50 objects in the context of the logged in user.
+        """Hide one or multiple objects in the context of the logged in user.
 
         :param thing_id: A single fullname or list of fullnames,
             representing objects which will be hidden.
-        :param _unhide: If True, unhide the object(s) instead.  Use
+        :param _unhide: If True, unhide the object(s) instead. Use
             :meth:`~praw.__init__.ReportMixin.unhide` rather than setting this
             manually.
 
         :returns: The json response from the server.
 
         """
-        if not isinstance(thing_id, six.string_types):
-            thing_id = ','.join(thing_id)
-        method = 'unhide' if _unhide else 'hide'
-        data = {'id': thing_id,
-                'executed': method}
-        response = self.request_json(self.config[method], data=data)
+        if isinstance(thing_id, six.string_types):
+            thing_id = [thing_id]
 
-        if self.user is not None:
-            self.evict(urljoin(self.user._url,  # pylint: disable=W0212
-                               'hidden'))
-        return response
+        id_chunks = chunk_sequence(thing_id, 50)
+        responses = []
+        for id_chunk in id_chunks:
+            id_chunk = ','.join(id_chunk)
+
+            method = 'unhide' if _unhide else 'hide'
+            data = {'id': id_chunk,
+                    'executed': method}
+
+            response = self.request_json(self.config[method], data=data)
+            responses.append(response)
+
+            if self.user is not None:
+                self.evict(urljoin(self.user._url,  # pylint: disable=W0212
+                                   'hidden'))
+        return responses
```

Comment:
Am I correct that this changes the response type always to a list? Can we make it such that it matches the input format (e.g., list in list out, single item in, single item out) so that this isn't a breaking change?